src/Undefinable/map.ts

```diff
@@ -11,9 +11,9 @@ import { Undefinable } from './Undefinable';
  *      * If you'd like return `Undefinable<*>` as `U`, use `andThen()`.
  *      * If the result of _selector_ is `undefined`, this throw an `Error`.
  */
-export function mapForUndefinable<T, U>(src: Undefinable<T>, selector: MapFn<T, U>): Undefinable<U> {
+export function mapForUndefinable<T, U>(src: Undefinable<T>, selector: MapFn<NotUndefined<T>, NotUndefined<U>>): Undefinable<U> {
```

Comment:
This change is nice to detect misusing `mapFor***()`. In such case, it should use `andThenFor***()`.